File: backend/ArBrain.FermentationControl.Domain/Entities/FermentationRecord.cs
using ArBrain.FermentationControl.Domain.Enums;

namespace ArBrain.FermentationControl.Domain.Entities;

public class FermentationRecord
{
    public Guid Id { get; set; }

    public Guid BeerId { get; set; }
    public Guid TankId { get; set; }

    public string BatchNumber { get; set; } = string.Empty;

    public decimal Temperature { get; set; }
    public decimal Ph { get; set; }
    public decimal Extract { get; set; }

    public DateTime RegisteredAt { get; set; }

    public string? Observations { get; set; }

    public FermentationStatus Status { get; set; }
}

File: frontend/src/models/FermentationRecord.ts
export type FermentationStatus = "InStandard" | "Warning" | "OutOfStandard";

export interface FermentationRecord {
  id: string;
  recordedAt: string;
  batchNumber: string;
  temperature: number;
  ph: number;
  extract: number;
  observation?: string | null;
  status: FermentationStatus;
  beerId: string;
  beerName: string;
  tankId: string;
  tankName: string;
}